### PR TITLE
LAA: thoroughly clarify stride-versioning code (NFC)

### PR DESCRIPTION
**Pull request by artagnon**
Thoroughly rewrite the comments and debug messages in the stride-versioning code in LoopAccessAnalaysis, learning from the recent a80dd4 (LAA: pre-commit tests for stride-versioning) that added tests to cover it.

### CODE REVIEW

File: llvm/lib/IR/ConstantRange.cpp [hunk 1]
```diff
@@ -440,6 +440,11 @@ bool ConstantRange::isAllNonNegative() const {
   return !isSignWrappedSet() && Lower.isNonNegative();
 }
 
+bool ConstantRange::isAllPositive() const {
+  // Empty and full set are automatically treated correctly.
+  return !isSignWrappedSet() && Lower.isStrictlyPositive();
+}
```
Comment (nikic) [inline llvm/lib/IR/ConstantRange.cpp:446]: Please split this out into a separate patch and add it to the exhaustive test in `TEST_F(ConstantRangeTest, Negative)`.

Comment (artagnon) [reply llvm/lib/IR/ConstantRange.cpp]: Thanks, fixing the issues in isAllPositive made me realize that this patch was wrong. Fixed now.

**Comment by fhahn** [comment]
Just to double check, the latest version doesn't seem to have any test changes (modulo adding new ones), is that intentional/expected?

**Comment by artagnon** [comment]
> Just to double check, the latest version doesn't seem to have any test changes (modulo adding new ones), is that intentional/expected?

Yes.

**Comment by artagnon** [comment]
Thanks to the comprehensive testing that was recently landed, we now know that the existing code is correct, and cannot be improved. The patch has now been converted into an NFC.

File: llvm/lib/Analysis/LoopAccessAnalysis.cpp [hunk 2]
```diff
@@ -2955,40 +2956,35 @@ void LoopAccessInfo::collectStridedAccess(Value *MemAccess) {
   // of various possible stride specializations, considering the alternatives
   // of using gather/scatters (if available).
 
-  const SCEV *MaxBTC = PSE->getSymbolicMaxBackedgeTakenCount();
+  const SCEV *BTC = PSE->getSymbolicMaxBackedgeTakenCount();
```
Comment (artagnon) [inline llvm/lib/Analysis/LoopAccessAnalysis.cpp:2959]: Small caveat: I think this should be `getBackedgeTakenCount`, the exact version, and not the symbolic-max version. However, it's not going to be easy to cook up a test to support this change, but I can think about it in a follow-up.

**Comment by artagnon** [comment]
Not pursuing this.
